rust-toolchain

```diff
@@ -0,0 +1 @@
+nightly-2023-09-05
```

Comment:
I'm curious to know why nightly Rust is required. It would be good to add some docs on this at some point.

Reply:
Yes, at this time it requires nightly Rust to compile. We started with stable Rust but at some point introduced some nightly-only features like "specialization". I think it is very easy to remove the dependency though - we can switch back to stable Rust later.